src/GraphQL/Types/TypeInformation.cs

```diff
@@ -206,7 +206,7 @@ private void ProcessTypeTree(List<(Type Type, NullabilityState Nullable)> typeTr
                     if (type.Type.IsGenericType)
                     {
                         var genericType = type.Type.GetGenericTypeDefinition();
-                        if (genericType == typeof(Task<>) || genericType == typeof(IDataLoaderResult<>))
+                        if (genericType == typeof(Task<>) || genericType == typeof(IDataLoaderResult<>) || genericType == typeof(IObservable<>))
```

Comment:
Plus ValueTask?

Reply:
That change exists in PR #2907

Reply:
(Returning a `ValueTask` is not supported by field resolvers at this point, so adding a condition for it here would be pointless.)